docs/building-in-chunks/batch-propose.md

```diff
@@ -0,0 +1,73 @@
+# Intro
+
+Design document for a variant of building in chunks with chained transaction support where we submit L2 blocks to L1 in batches. This is based on [this suggestion](https://github.com/AztecProtocol/engineering-designs/pull/70#pullrequestreview-3019840618) from @LHerskind.
+
+## Overview
+
+We define an L1 **checkpoint** ("frame" in the original proposal) as a new period of time, composed by multiple slots. Within a checkpoint, the proposer is always the same. The `Rollup.propose` function now accepts multiple L2 blocks instead of one, up to one for each slot. Only the attestations for the last block in the checkpoint are posted and validated, since any invalid block within the checkpoint would invalidate all subsequent blocks. The batch requires a mapping from blob index to block within the batch.
+
+This requires no circuit changes, and requires changes in the node related to building in chunks.
+
+## Numbers
+
+### Checkpoint and slot durations
+
+Since we are not making changes to circuits, each block still takes up at least one blob. If we target 6 blobs per tx, which may reduce our chances at L1 inclusion, we can squeeze no more than 6 slots per checkpoint. Assuming 1200 bytes per tx effect, each 128kb blob can carry about 100 txs. Assuming 10 TPS, we require 12kb per second for DA, which is 1 blob every 10.6 seconds, roughly 1 blob per L1 slot. This means the duration of a checkpoint cannot be more than 6 L1 slots (72s), and the duration of each L2 slot should be equivalent to one L1 slot (12s).
+
+Note that the above assumes 10TPS. If we go below this, then each block would not fully utilize its blob, leading to wasted blobspace and increased DA cost per tx. We should expect proposers to **not** produce a block for a slot if they have not filled the block, and instead accumulate txs into the following block in the slot within their checkpoint. This leads to less DA cost, which is borne by the proposer, but worse UX, since users now have to wait longer for L2 inclusion.
+
+If we land at lower TPS or less bytes per tx, this can lead to worse UX overall, since a proposer would be incentivized to only post a single block within their checkpoint, effectively increasing block times to 72s. If we assume 300 bytes per tx, which is the current lower bound for a transfer with out-of-band messaging (which should be the preference for app developers who want to reduce tx costs) at 6 TPS, a single blob takes the full 72s to be filled.
+
+On the other hand, if we consider 2400 bytes per tx effect, which is the size of a transfer with fpc support with all notes posted on-chain, then each blob can carry about 50 txs only. Assuming 10 TPS, this means we fill 2 blobs per L1 slot, so checkpoints cannot be longer than 3 L1 slots (36s).
```

Comment:
Why is this so high? The private refund flow is not needed here. This should be max 3 notes, and one is transient I beleive.